### PR TITLE
V4 back

### PR DESCRIPTION
**Pull request by rocambille**

### CODE REVIEW

**Comment by GorskiAnthony** [comment]
Why set the backend port to 6000? 
On mac it tells me `net::ERR_UNSAFE_PORT`.

Otherwise, I validate 😄

**Review by GorskiAnthony — APPROVED** [review]
Good for me 👍

**Comment by rocambille** [comment]
> Why set the backend port to 6000?
> On mac it tells me `net::ERR_UNSAFE_PORT`.

@GorskiAnthony because the 5000 is often reserved on mac, and the 6000 is sometimes banned by chrome... I think whatever the port we choose, we always have an issue somewhere :sweat_smile: 

maybe we can give a try to the 8000 ?

**Comment by GorskiAnthony** [comment]
Or 5500 💪

**Comment by GorskiAnthony** [comment]
> 
> 
> 
> > Why set the backend port to 6000?
> > On mac it tells me `net::ERR_UNSAFE_PORT`.
> 
> @GorskiAnthony because the 5000 is often reserved on mac, and the 6000 is sometimes banned by chrome... I think whatever the port we choose, we always have an issue somewhere :sweat_smile: 
> 
> maybe we can give a try to the 8000 ?

Or 5500 ?

**Comment by rocambille** [comment]
> Or 5500 ?

I checked the known reserved ports : https://en.wikipedia.org/wiki/List_of_TCP_and_UDP_port_numbers

I ended up with the 3310 to stay in the 3000 range like the front (and with a ref I'm sure you get @GorskiAnthony ;) )

File: backend/.env.sample [hunk 1]
```diff
@@ -1,7 +1,8 @@
-APP_PORT=6000
-FRONTEND_URL=http://localhost:3000
+APP_PORT=3310
+APP_SECRET=eac05f7808f3d98a76614aceac6f6ca8
```
Comment (GorskiAnthony) [inline backend/.env.sample]: What is APP_SECRET used for?

File: backend/src/tables.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+const tables = {};
+
+/* ************************************************************************* */
+// that's where you should register your managers
+/* ************************************************************************* */
+
+const ItemManager = require("./models/ItemManager");
+
+tables.item = new ItemManager();
+
```
Comment (GorskiAnthony) [inline backend/src/tables.js]: To optimize code, imports, etc.

```js
const ManagersConfig = [
  { name: "item", managerClass:require("./models/ItemManager") },
  // Add other managers here
];

ManagersConfig.forEach(({ name, managerClass }) => {
  const manager = new managerClass();
  tables[name] = manager;
});
```

File: backend/src/models/ItemManager.js [hunk 3]
```diff
@@ -5,18 +5,41 @@ class ItemManager extends AbstractManager {
...
+    );
+
+    return result.insertId;
   }
 
-  update(item) {
-    return this.database.query(
-      `update ${this.table} set title = ? where id = ?`,
-      [item.title, item.id]
+  // the Rs of CRUD
```
Comment (GorskiAnthony) [inline backend/src/models/ItemManager.js]: Rs or R 😏 ?

**Review by GorskiAnthony — APPROVED** [review]

**Pull request merged**